### DISCUSSION TITLE
invalid file name character replacement in recent versions

### DISCUSSION THREAD

**Discussion by Codereamp** [gildas-lormeau/SingleFile#1632] [Q&A] [answered]
Hi,

I noticed that there's a new feature by default listed as `Added support of full-width UTF-8 symbols to replace invalid file name characters (e.g ">", ":", "|")`  (probably [the commit](https://github.com/gildas-lormeau/SingleFile/commit/b84a36b3628906b09d73cfa5a44c48d6ee659226)). So when formatting the time, for example, colon (:) is replaced with 'full width colon' (unicode point ff1a). 

Is there a existing way force it to return to underscore usage? The reasons why it may be benefifical

- A font one might use can handle such unicode points badly. An example: In Windows 10, Consolas font (bundled with Windows) is often used for console application and I just tested that in cmd.exe when showing filenames with dir command the file name saved by SingleFile has question mark in boxes instead of colons. 
- In my case I have some legacy code that works only with ansi names (one one-byte encodings), so I carefully try to avoid unicode symbols that is not decoded correctly to the current one-byte encoding codepage

Thanks

**Comment by gildas-lormeau** [comment, marked as answer]
The behavior has indeed changed, see #1614.

To revert the changes, update the the hidden option `filenameReplacementCharacters` (see https://github.com/gildas-lormeau/SingleFile/wiki/Hidden-options) by replacing:

`"filenameReplacementCharacters": ["～", "＋", "？", "％", "＊", "：", "｜", "＂", "＜", "＞", "＼"],`

with:

`"filenameReplacementCharacters": ["_", "_", "_", "_", "_", "_", "_", "_", "_", "_", "_"],`

**Reply by Codereamp** [reply]
Thanks for the suggestion, unfortunately in ff 134.0b1 SingleFile version 1.22.73 it didn't work. What I did and observed

- Your suggestion verbatim array to array (11 undescores)
- Tried to match the number of elements (13 undescores matching the number of array elements in filenameReplacedCharacters instead of 11 in your post). 
- disabled,enabled SingleFile
- Restarted the ff profile
- Looked at the re-export of the imported settings to check whether the new values are there. Yes, they exist in the re-exported variant
- In a profile previously lacking SingleFile, installed the latest version, made export->change (11 underscores)->import

**Reply by gildas-lormeau** [reply]
The extension is not up to date on the Mozilla store as I write this reply. It will be available once the code has been reviewed by Mozilla.

**Reply by gildas-lormeau** [reply]
The update is now available on the store of Mozilla.

**Reply by Codereamp** [reply]
@gildas-lormeau, thanks, the setting works as expected

**Reply by gildas-lormeau** [reply]
Thank you for the feedback!

**Comment by Codereamp** [comment]
Sorry, I'm here again :)

As a dedicated fun of SingleFile, I'm using it everywhere, including mobile Firefox :)
Just noticed that on Android (11) the default filename with full-width characters are not rendered correctly (see the screenshot fragment from the MixPlorer file list)

<img src="https://github.com/user-attachments/assets/51b10c1c-f782-4946-8ad2-004193a1179d" width="500">



The same effect is in the default Redmi File Manager.  

This is just to add to the statistics of this new feature.

But I'm writing also because I could not manage to change the setting (the mobile version of the extension is 1.22.75, updated on Dec 10). The saved file still contains full-width symbols and the re-exported settings  file (saved for checking immediately after import) contains reversed `filenameReplacementCharacters` symbols, so not many underscores as I used for the replace, but the original ones.  Not a big deal, but maybe there's an explanation

Thanks
